### PR TITLE
elfio: Add version 3.11

### PR DESCRIPTION
**Pull request by serge1**
Specify library name and version:  **elfio/3.11**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/elfio/all/conanfile.py [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
+            check_min_cppstd(self, 14)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("elfio-{}".format(self.version), self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version])
+        rename(self, "elfio-{}".format(self.version), "elfio")
```
Comment (uilianries) [inline recipes/elfio/all/conanfile.py:27]:
```suggestion
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```
Strip root parameter solves the situation

File: recipes/elfio/all/conanfile.py [hunk 2]
```diff
@@ -1,32 +1,35 @@
...
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def configure(self):
         if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
+            check_min_cppstd(self, 14)
```
Comment (uilianries) [inline recipes/elfio/all/conanfile.py:20]: Does it require C++14 since 3.11? If yes, it would need a condition, based on the package version.

Comment (serge1) [reply recipes/elfio/all/conanfile.py]: > Does it require C++14 since 3.11?

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**